br/pkg/lightning/config/config.go

```diff
@@ -187,7 +187,7 @@ func (d *DBStore) adjust(
 		if d.Security.TLSConfig == nil {
 			/* #nosec G402 */
 			d.Security.TLSConfig = &tls.Config{
-				MinVersion:         tls.VersionTLS10,
+				MinVersion:         tls.VersionTLS12,
```

Comment:
dumpling and lightning may connect to older version of TiDB and MySQL, and their TLS version is older than v1.2. Should we support this case @frank945946

Reply:
TLSv1.2 is introduced in year 2008, 15 years ago, that's old enough.
And as long as those software support TLSv1.2 above, change this MinVersion will not cause any trouble.
Without doubt, that's the case for TiDB.

And then I check the release history of MySQL ...
```
Release	General availability	Latest release
5.5 LTS	3 December 2010	2018-10-22
5.6 LTS	5 February 2013	2021-01-20
5.7 LTS	21 October 2015	2023-10-25
8.0 LTS	19 April 2018	2023-10-25
```

I'm sure it's pretty safe to do this change according to the above information.

Reply:
It is not so 'safe' actually. We do know many TiDB users still using old clients. However, it is reasonable to remove the support of 'TLS 1.0/1.1'.

Thanks @lance6716

Reply:
Removing the support of 'TLS 1.0/1.1'  LTGM  and please leave a note to the corresponding docs @lance6716 .